piccolo_api/audit_logs/tables.py

```diff
@@ -0,0 +1,111 @@
+import typing as t
+import uuid
+from datetime import datetime
+from enum import Enum
+
+from piccolo.apps.user.tables import BaseUser
+from piccolo.columns import Text, Timestamp, Varchar
+from piccolo.table import Table
+
+
+class AuditLog(Table):
+    class ActionType(str, Enum):
+        """An enumeration of AuditLog table actions type."""
+
+        CREATING = "CREATING"
+        UPDATING = "UPDATING"
+        DELETING = "DELETING"
+
+    action_time = Timestamp()
+    action_type = Varchar(choices=ActionType)
+    action_user = Varchar()
+    change_message = Text()
+
+    @classmethod
+    async def post_save_action(cls, table: t.Type[Table], user_id: int):
+        """
+        A method for tracking creating record actions.
+
+        :param table:
+            A table for which we monitor activities.
+        :param user_id:
+            The ``primary key`` of authenticated user.
+        """
+        result = cls(
+            action_time=datetime.now(),
+            action_type=cls.ActionType.CREATING,
+            action_user=cls.get_user_username(user_id),
+            change_message=f"User {cls.get_user_username(user_id)} "
+            f"create new row in {table._meta.tablename.title()} table",
+        )
+        await result.save().run()
+
+    @classmethod
+    async def post_patch_action(
```

Comment:
I'd consider naming these methods slightly differently. I know what you mean by them, and they are correct. But having `post_patch` could be confusing - someone might misread it as being about posts and not patches.

Instead of `post_`, how about just `record_`? So it would be `record_patch_action`?

We will never have a `pre_` version, so no need to distinguish with `post_`.

Reply:
You're right. I will change that.